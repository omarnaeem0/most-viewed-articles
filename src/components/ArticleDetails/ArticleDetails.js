import classNames from "classnames";
import Text from "../Text";
import styles from "./styles.module.scss";

function ArticleDetails({ data, setSelectedArticle }) {
  const image = data?.media?.[0]?.["media-metadata"]?.[2];

  return (
    <div className={styles.container}>
      <button
        className={styles.backButton}
        onClick={() => setSelectedArticle(null)}
      >
        <Text variant={"paragraph"}>Go back</Text>
      </button>
      <Text variant={"heading"}>{data.title}</Text>
      <Text variant={"title"} className={styles.marginBottom}>
        {data.abstract}
      </Text>

      <img className={styles.image} src={image.url} />
      <span className={styles.imageDetails}>
        <Text
          variant={"paragraph"}
          className={classNames(styles.caption, styles.inline)}
        >
          {data.media?.[0].caption}{" "}
        </Text>
        <Text variant={"paragraph"} className={styles.inline}>
          Pool photo by {data.media?.[0].copyright}
        </Text>
      </span>
      <div className={styles.articleLinkContainer}>
        <a className={styles.articleLink} href={data.url}>See full article</a>
      </div>
    </div>
  );
}

export default ArticleDetails;
