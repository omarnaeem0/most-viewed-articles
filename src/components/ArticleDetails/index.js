export {default} from './ArticleDetails.container';
