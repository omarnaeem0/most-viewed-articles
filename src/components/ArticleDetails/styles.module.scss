.container {
  max-width: 600px;
  margin: 0px auto;
  text-align: start;
}

.backButton {
  border: none;
  background-color: transparent;
  text-decoration: underline;
  cursor: pointer;
  margin: 12px 0px;
}

.image {
  width: 600px;
  height: auto;
}

.imageDetails {
  margin: 10px 20px 0 0;
}

.caption {
  text-align: left;
  font-size: 0.9375rem;
  line-height: 1.25rem;
}

.marginBottom {
  margin-bottom: 24px;
}

.inline {
  display: inline;
}

.articleLinkContainer {
  background: var(--color-background-primary, #FFFFFF);
  border: 1px solid var(--color-stroke-quaternary, #DFDFDF);
  cursor: pointer;
  margin: 24px 0px;
  text-align: center;
  display: flex;
}

.articleLink {
  color: var(--color-content-primary, #121212);
  box-shadow: none;
  text-decoration: none !important;
  padding: 24px;
  width: 100%;
  flex: 1
}
