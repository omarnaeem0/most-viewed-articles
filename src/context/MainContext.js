import { createContext } from "react";

export const MainContext = createContext({
  results: [],
  setResults: () => {},
  selectedArticle: null,
  setSelectedArticle: () => {},
});
