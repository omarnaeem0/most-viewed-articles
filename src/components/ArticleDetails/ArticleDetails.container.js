import { useContext } from "react";
import ArticleDetails from "./ArticleDetails";
import { MainContext } from "../../context/MainContext";

function ArticleDetailsContainer() {
  const { results, selectedArticle, setSelectedArticle } =
    useContext(MainContext);
  if (selectedArticle === null) return null;
  const data = results[selectedArticle];
  
  return <ArticleDetails data={data} setSelectedArticle={setSelectedArticle} />;
}

export default ArticleDetailsContainer;
