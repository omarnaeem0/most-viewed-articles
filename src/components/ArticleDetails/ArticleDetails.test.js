import { render, screen, fireEvent } from "@testing-library/react";
import ArticleDetails from "./ArticleDetails";
import renderer from "react-test-renderer";
import { ContextWrapper } from "../../context/MainContext";
import ArticleDetailsContainer from ".";

const data = {
  uri: "nyt://article/5b3ac615-1536-52b4-a1c3-ea3931b189e8",
  url: "https://www.nytimes.com/2024/03/27/well/colon-cancer-symptoms-treatment.html",
  id: 100000009368097,
  asset_id: 100000009368097,
  source: "New York Times",
  published_date: "2024-03-27",
  updated: "2024-03-28 09:33:37",
  section: "Well",
  subsection: "",
  nytdsection: "well",
  adx_keywords:
    "Colon and Colorectal Cancer;Colonoscopy;Disease Rates;Digestive Tract;Tests (Medical);Content Type: Service;Youth;Diet and Nutrition;American Cancer Society;United States",
  column: null,
  byline: "By Knvul Sheikh",
  type: "Article",
  title: "More Young People Than Ever Will Get Colorectal Cancer This Year",
  abstract:
    "Colon and rectal cancers are increasing among people younger than 50. Experts have a few ideas about why.",
  des_facet: [
    "Colon and Colorectal Cancer",
    "Colonoscopy",
    "Disease Rates",
    "Digestive Tract",
    "Tests (Medical)",
    "Content Type: Service",
    "Youth",
    "Diet and Nutrition",
  ],
  org_facet: ["American Cancer Society"],
  per_facet: [],
  geo_facet: ["United States"],
  media: [
    {
      type: "image",
      subtype: "photo",
      caption: "",
      copyright: "Jeannie Phan",
      approved_for_syndication: 1,
      "media-metadata": [
        {
          url: "https://static01.nyt.com/images/2024/03/27/well/27Well-Colon-Cancer/27Well-Colon-Cancer-thumbStandard.jpg",
          format: "Standard Thumbnail",
          height: 75,
          width: 75,
        },
        {
          url: "https://static01.nyt.com/images/2024/03/27/well/27Well-Colon-Cancer/27Well-Colon-Cancer-mediumThreeByTwo210.jpg",
          format: "mediumThreeByTwo210",
          height: 140,
          width: 210,
        },
        {
          url: "https://static01.nyt.com/images/2024/03/27/well/27Well-Colon-Cancer/27Well-Colon-Cancer-mediumThreeByTwo440.jpg",
          format: "mediumThreeByTwo440",
          height: 293,
          width: 440,
        },
      ],
    },
  ],
  eta_id: 0,
};

test("ArticleDetails component with mock data", () => {
  const tree = renderer.create(<ArticleDetails data={data} />).toJSON();
  expect(tree).toMatchSnapshot();
});
test("ArticleDetails component with mock data matching data", () => {
  render(
    <ArticleDetails
      data={data}
      setSelectedArticle={(i) => expect(i).toBe(null)}
    />
  );
  expect(screen.getByRole("img")).toHaveAttribute(
    "src",
    data.media[0]["media-metadata"][2].url
  );
  expect(screen.getByRole("link")).toHaveAttribute("href", data.url);
  expect(screen.getByText(/See full article/i)).toHaveAttribute(
    "href",
    data.url
  );
  const buttonElement = screen.getByRole("button");
  fireEvent.click(buttonElement);

  expect(buttonElement).toBeDefined();
});
test("ArticleDetailsContainer with contextWrapper with default props", () => {
  const tree = renderer
    .create(
      <ContextWrapper>
        <ArticleDetailsContainer />
      </ContextWrapper>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test("ArticleDetailsContainer snapshot returns details", () => {
  const tree = renderer
    .create(
      <ContextWrapper defaultResults={[data]} defaultSelectedArticle={0}>
        <ArticleDetailsContainer />
      </ContextWrapper>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test("ArticleDetailsContainer snapshot returns null", () => {
  const tree = renderer
    .create(
      <ContextWrapper defaultResults={[data]} defaultSelectedArticle={null}>
        <ArticleDetailsContainer />
      </ContextWrapper>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
