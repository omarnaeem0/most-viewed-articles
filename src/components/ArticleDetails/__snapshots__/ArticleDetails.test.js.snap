// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleDetails component with mock data 1`] = `
<div
  className="container"
>
  <button
    className="backButton"
    onClick={[Function]}
  >
    <p
      className="paragraph"
      data-testid="text-paragraph"
    >
      Go back
    </p>
  </button>
  <p
    className="heading"
    data-testid="text-heading"
  >
    More Young People Than Ever Will Get Colorectal Cancer This Year
  </p>
  <p
    className="title marginBottom"
    data-testid="text-title"
  >
    Colon and rectal cancers are increasing among people younger than 50. Experts have a few ideas about why.
  </p>
  <img
    className="image"
    src="https://static01.nyt.com/images/2024/03/27/well/27Well-Colon-Cancer/27Well-Colon-Cancer-mediumThreeByTwo440.jpg"
  />
  <span
    className="imageDetails"
  >
    <p
      className="paragraph caption inline"
      data-testid="text-paragraph"
    >
       
    </p>
    <p
      className="paragraph inline"
      data-testid="text-paragraph"
    >
      Pool photo by 
      Jeannie Phan
    </p>
  </span>
  <div
    className="articleLinkContainer"
  >
    <a
      className="articleLink"
      href="https://www.nytimes.com/2024/03/27/well/colon-cancer-symptoms-treatment.html"
    >
      See full article
    </a>
  </div>
</div>
`;

exports[`ArticleDetailsContainer snapshot returns details 1`] = `
<div
  className="container"
>
  <button
    className="backButton"
    onClick={[Function]}
  >
    <p
      className="paragraph"
      data-testid="text-paragraph"
    >
      Go back
    </p>
  </button>
  <p
    className="heading"
    data-testid="text-heading"
  >
    More Young People Than Ever Will Get Colorectal Cancer This Year
  </p>
  <p
    className="title marginBottom"
    data-testid="text-title"
  >
    Colon and rectal cancers are increasing among people younger than 50. Experts have a few ideas about why.
  </p>
  <img
    className="image"
    src="https://static01.nyt.com/images/2024/03/27/well/27Well-Colon-Cancer/27Well-Colon-Cancer-mediumThreeByTwo440.jpg"
  />
  <span
    className="imageDetails"
  >
    <p
      className="paragraph caption inline"
      data-testid="text-paragraph"
    >
       
    </p>
    <p
      className="paragraph inline"
      data-testid="text-paragraph"
    >
      Pool photo by 
      Jeannie Phan
    </p>
  </span>
  <div
    className="articleLinkContainer"
  >
    <a
      className="articleLink"
      href="https://www.nytimes.com/2024/03/27/well/colon-cancer-symptoms-treatment.html"
    >
      See full article
    </a>
  </div>
</div>
`;

exports[`ArticleDetailsContainer snapshot returns null 1`] = `null`;

exports[`ArticleDetailsContainer with contextWrapper with default props 1`] = `null`;
